### PR TITLE
Added 10g possibility

### PR DESCRIPTION
**Pull request by madams51703**
Added alter_database into mix.  Fixed a little grammar

### CODE REVIEW

**Comment by teverett** [comment]
@madams51703 thanks

File: plsql/PlSqlLexer.g4 [hunk 1]
```diff
@@ -2266,7 +2276,7 @@ APPROXIMATE_NUM_LIT: FLOAT_FRAGMENT ('E' ('+'|'-')? (FLOAT_FRAGMENT | [0-9]+))?
 
 // Rule #--- <CHAR_STRING> is a base for Rule #065 <char_string_lit> , it incorporates <character_representation>
 // and a superfluous subtoken typecasting of the "QUOTE"
-CHAR_STRING: '\'' (~('\'' | '\r' | '\n') | '\'' '\'' | NEWLINE)* '\'';
+CHAR_STRING: '\''  (~('\'' | '\r' | '\n') | '\'' '\'' | NEWLINE)* '\'';
```
Comment (KvanTTT) [inline plsql/PlSqlLexer.g4]: Excess whitespace.

File: plsql/PlSqlParser.g4 [hunk 2]
```diff
@@ -1454,11 +1457,12 @@ tempfile_specification
     ;
 
 datafile_tempfile_spec
-    : CHAR_STRING? (SIZE size_clause)? REUSE? autoextend_clause?
+    : (CHAR_STRING)? (SIZE size_clause)? REUSE? autoextend_clause?
```
Comment (KvanTTT) [inline plsql/PlSqlParser.g4]: Excess parenthesis.

File: plsql/PlSqlParser.g4 [hunk 3]
```diff
@@ -2125,7 +2129,7 @@ cache_or_nocache
     ;
 
 database_name
-    : regular_id
+    : regular_id
```
Comment (KvanTTT) [inline plsql/PlSqlParser.g4]: Excess whitespace.

File: plsql/PlSqlParser.g4 [hunk 4]
```diff
@@ -2226,7 +2240,7 @@ create_datafile_clause
     ;
 
 alter_datafile_clause
-    : DATAFILE (','? filename|filenumber)+ 
+    : DATAFILE (','? (filename|filenumber) )+
```
Comment (KvanTTT) [inline plsql/PlSqlParser.g4]: Please insert whitespace before and after the vertical bar.

File: plsql/Java/PlSqlBaseParser.java [hunk 5]
```diff
@@ -15,4 +16,12 @@ public boolean isVersion12() {
...
         _isVersion12 = value;
     }
+
+    public boolean isVersion10() {
+        return _isVersion10;
+    }
+
+    public void setVersion10(boolean value) {
+        _isVersion10 = value;
+    }
```
Comment (KvanTTT) [inline plsql/Java/PlSqlBaseParser.java]: Could you also make corresponding changes to C# PlSqlBaseParser.cs?

**Comment by madams51703** [comment]
Sure, I can do that.

**Pull request merged**